tests/blockchain/eth/interfaces/test_solidity_compiler.py

```diff
@@ -14,14 +14,54 @@
 You should have received a copy of the GNU Affero General Public License
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
-
+import os
+from os.path import dirname, abspath
 
 from nucypher.blockchain.eth.deployers import NucypherTokenDeployer
+from nucypher.blockchain.eth.sol.compile import SolidityCompiler, SourceDirs
+from nucypher.utilities.sandbox.blockchain import TesterBlockchain
 
 
 def test_nucypher_contract_compiled(testerchain, test_registry):
     # Ensure that solidity smart contacts are available, post-compile.
     origin, *everybody_else = testerchain.client.accounts
 
     token_contract_identifier = NucypherTokenDeployer(registry=test_registry, deployer_address=origin).contract_name
-    assert token_contract_identifier in testerchain._BlockchainDeployerInterface__raw_contract_cache
+    assert token_contract_identifier in testerchain._raw_contract_cache
+    token_data = testerchain._raw_contract_cache[token_contract_identifier]
+    assert len(token_data) == 1
+    assert "v0.0.0" in token_data
+
+
+def test_multi_source_compilation(testerchain):
+    solidity_compiler = SolidityCompiler(source_dirs=[
+        (SolidityCompiler.default_contract_dir(), None),
+        (SolidityCompiler.default_contract_dir(), {TesterBlockchain.TEST_CONTRACTS_DIR})
+    ])
+    interfaces = solidity_compiler.compile()
+
+    # Remove AST because id in tree node depends on compilation scope
```

Comment:
Woah - can you expand a bit on this?

Reply:
Each element in AST has ID field. This value depends on scope of compiled files (all contracts). First source for SolidityCompiler includes only main contracts, second - also tests. Merging two results lead that main contracts has AST from first compilation, test contracts from the second. Interface has contracts with the same AST as in second compilation for all contracts. We don't use AST yet and main part for comparing is bytecode, so I remove it